src/cli/argument-parser.js

```diff
@@ -0,0 +1,214 @@
+import { readSync as read } from 'read-file-relative';
+import { MESSAGE, getText } from '../messages';
+import getViewPortWidth from '../utils/get-viewport-width';
+import wordwrap from '../utils/word-wrap';
+
+var ensureDir = promisify(mkdirp);
+
+const REMOTE_ALIAS_RE = /^(\d*)remote$/;
+
+const DESCRIPTION = dedent`
+    In the browser list, you can use aliases (e.g. "ie9", "chrome", etc.) as well as paths to executables.
+
+    To run tests against all installed browsers, use the "all" alias.
+
+    To use a remote browser connection (e.g. to connect a mobile device), specify "remote" as a browser alias. If you
+    need to connect multiple devices, prefix the alias with the number of browsers you want to connect (e.g., "3remote").
+
+    You can use one or more file paths or glob patterns to specify which tests to run.
+
+    More info: http://testcafe.devexpress.com/Documentation/CLI
+`;
+
+export default class CliArgumentParser {
+    constructor () {
+        this.program = new Command('testcafe');
+
+        this.cwd = process.cwd();
+
+        this.src             = null;
+        this.browsers        = null;
+        this.filter          = null;
+        this.remoteCount     = 0;
+        this.reportOutStream = null;
+        this.opts            = null;
+
+        this._describeProgram();
+    }
+
+    static _parsePortNumber (value) {
+        var number = parseInt(value, 10);
+
+        if (isNaN(value))
+            throw new Error(getText(MESSAGE.portNumberIsNotInteger));
+
+        return number;
+    }
+
+    static _optionValueToRegExp (name, value) {
+        if (value === void 0)
+            return value;
+
+        try {
+            return new RegExp(value);
+        }
+        catch (err) {
+            throw new Error(getText(MESSAGE.optionValueIsNotValidRegExp, name));
+        }
+    }
+
+    static _replaceAllBrowsersAlias (browserList, browser, allAliases) {
+        if (browser === 'all')
+            browserList = browserList.concat(allAliases);
+
+        else
+            browserList.push(browser);
+
+        return browserList;
+    }
+
+    static _getDescription () {
+        // NOTE: add empty line to workaround commander-forced indentation on the first line.
+        return '\n' + wordwrap(DESCRIPTION, 2, getViewPortWidth(true));
+    }
+
+    _describeProgram () {
+        this.program
+
+            .version(JSON.parse(read('../../package.json')).version)
+            .usage('[options] <comma-separated-browser-list> <file-or-glob ...>')
+            .description(CliArgumentParser._getDescription())
+
+            .option('-b, --list-browsers', 'output the available browser aliases')
+            .option('-r, --reporter <name>', 'specify the reporter type to use')
+            .option('-p, --report-path <path>', 'specify the path to which the report file should be saved')
+            .option('-s, --screenshots <path>', 'enable screenshot capturing and specify the path to save them to')
```

Comment:
oops... 'them' doesn't relate ro anything anymore...
`enable screenshot capturing and specify the path to save the screenshots to`